support/sys-socket.c

```diff
@@ -0,0 +1,139 @@
+/*
+ * <sys/socket.h> wrapper functions.
+ *
+ * Authors:
+ *   Steffen Kiess (s-kiess@web.de)
+ *
+ * Copyright (C) 2015 Steffen Kiess
+ */
+
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <sys/un.h>
+
+#include <stddef.h>
+
+#include "map.h"
+#include "mph.h"
+
+G_BEGIN_DECLS
+
+int
+Mono_Posix_Syscall_socketpair (int domain, int type, int protocol, int* sv0, int* sv1)
```

Comment:
If this took a `int *filedes`, you wouldn't need `int filedes[2]` in the body... ;-)

Which answers a previous question: we should standardize on the `int[]` `socketpair()` overload, removing the `(int, int)` overload.